libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -412,24 +418,38 @@ void AP_DDS_Client::on_topic (uxrSession* uxr_session, uxrObjectId object_id, ui
     */
 
     switch (object_id.id) {
-    case topics[to_underlying(TopicIndex::JOY_SUB)].dr_id.id:
-        sensor_msgs_msg_Joy topic;
-        const bool success = sensor_msgs_msg_Joy_deserialize_topic(ub, &topic);
+    case topics[to_underlying(TopicIndex::JOY_SUB)].dr_id.id: {
+        const bool success = sensor_msgs_msg_Joy_deserialize_topic(ub, &rx_joy_topic);
 
         if (success == false) {
             break;
         }
 
         uint32_t* count_ptr = (uint32_t*) args;
         (*count_ptr)++;
-        if (topic.axes_size >= 4) {
+        if (rx_joy_topic.axes_size >= 4) {
             GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received sensor_msgs/Joy: %f, %f, %f, %f",
-                          topic.axes[0], topic.axes[1], topic.axes[2], topic.axes[3]);
+                          rx_joy_topic.axes[0], rx_joy_topic.axes[1], rx_joy_topic.axes[2], rx_joy_topic.axes[3]);
         } else {
             GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received sensor_msgs/Joy: Insufficient axes size ");
         }
         break;
     }
+    case topics[to_underlying(TopicIndex::DYNAMIC_TRANSFORMS_SUB)].dr_id.id: {
+        const bool success = tf2_msgs_msg_TFMessage_deserialize_topic(ub, &rx_dynamic_transforms_topic);
+        if (success == false) {
+            break;
+        }
+        uint32_t* count_ptr = (uint32_t*) args;
+        (*count_ptr)++;
+        if (rx_dynamic_transforms_topic.transforms_size > 0) {
+            GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received tf2_msgs/TFMessage of length: %zu",
```

Comment:
Apparently the Ardupilot C compiler doesn't support `%z`. It failed CI and I also wasn't able to compile it locally.

Reply:
Which board is it failing for you on?

Reply:
Pixhawk4

Reply:
With the last commit, it passes CI with the use of a static cast. 
Please approve if you are happy.